pkgs/by-name/ra/raylib/package.nix

```diff
@@ -1,85 +1,80 @@
-{ stdenv
-, lib
-, fetchFromGitHub
-, cmake
-, fetchpatch
-, mesa
-, libGLU
-, glfw
-, libX11
-, libXi
-, libXcursor
-, libXrandr
-, libXinerama
-, alsaSupport ? stdenv.hostPlatform.isLinux
-, alsa-lib
-, pulseSupport ? stdenv.hostPlatform.isLinux
-, libpulseaudio
-, sharedLib ? true
-, includeEverything ? true
-, raylib-games
-, darwin
-, autoPatchelfHook
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  autoPatchelfHook,
+  glfw,
+  SDL2,
+  alsa-lib,
+  libpulseaudio,
+  raylib-games,
+  platform ? "Desktop", # Note that "Web", "Android" and "Raspberry Pi" do not currently work
+  pulseSupport ? stdenv.hostPlatform.isLinux,
+  alsaSupport ? false,
+  sharedLib ? true,
+  includeEverything ? true,
 }:
 let
-  inherit (darwin.apple_sdk.frameworks) Carbon Cocoa OpenGL;
-in
-stdenv.mkDerivation (finalAttrs: {
-  pname = "raylib";
-  version = "5.0";
+  inherit (lib) optional;
 
-  src = fetchFromGitHub {
-    owner = "raysan5";
-    repo = "raylib";
-    rev = finalAttrs.version;
-    hash = "sha256-gEstNs3huQ1uikVXOW4uoYnIDr5l8O9jgZRTX1mkRww=";
-  };
+  pname = "raylib";
+in
 
-  nativeBuildInputs = [
-    cmake
-  ] ++ lib.optional stdenv.hostPlatform.isLinux autoPatchelfHook;
+lib.checkListOfEnum "${pname}: platform"
+  [
+    "Desktop"
+    "Web"
+    "Android"
+    "Raspberry Pi"
+    "SDL"
+  ]
+  [ platform ]
+  (
+    stdenv.mkDerivation (finalAttrs: {
+      __structuredAttrs = true;
 
-  buildInputs = [ glfw ]
-    ++ lib.optionals stdenv.hostPlatform.isLinux [ mesa libXi libXcursor libXrandr libXinerama ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [ Carbon Cocoa ]
-    ++ lib.optional alsaSupport alsa-lib
-    ++ lib.optional pulseSupport libpulseaudio;
+      inherit pname;
+      version = "5.5";
 
-  propagatedBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [ libGLU libX11 ]
```

Comment:
Why not propagate `[ libGLU libX11 ]`?

Reply:
The better question is, *why* propagate those libraries? They are optional dependencies of Raylib that should only be used if Raylib is missing functionality for something, they are *not required* for building Raylib applications.

Reply:
I'll answer your question then. Propagate those libraries because they are required for the majority, if not all, downstream consumers of Raylib. If they are not used downstream, then they are not part of the runtime closure.

Reply:
What I mean is, propagating `libX11` will slow down the setup of builds that don't require it, but if the depender `foo` of `raylib` doesn't use `libX11`, then it won't propagate `libX11` as aa runtime dependency of `foo`. That's not the case if the depender detects `libX11` somehow and pulls it in, in which case that is the correct thing to do anyway.

I don't want to block this PR on this point but I'm not going to be the one to hit the merge button.

Reply:
How does Nix determine whether the depender uses libX11 or not?

I'm open to changing this as well, I just don't agree with propagating optional libraries. However removing them might cause a headache for a few people for whom it's a breaking change, so doing that isn't ideal anyway.

Reply:
It's barely [documented](https://nix.dev/manual/nix/2.25/introduction.html#complete-dependencies) but when the build is completed all the files in the outputs are scanned for references to the input paths. It sounds crazy but it works.

Reply:
Alright, I'm fine to revert this then. Though I still don't fully agree with it.